packages/frontend2/src/app/_components/nav/MobileNavContext.tsx

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { createContext, useContext, useState } from 'react'
+
+const MobileNavContext = createContext<{
+  open: boolean
+  setOpen: (open: boolean) => void
+} | null>(null)
+
+export function MobileNavProvider({ children }: { children: React.ReactNode }) {
+  const [open, setOpen] = useState(false)
+
+  return (
+    <MobileNavContext.Provider value={{ open, setOpen }}>
+      {children}
+    </MobileNavContext.Provider>
+  )
+}
+
+export function useMobileNav() {
+  const context = useContext(MobileNavContext)
+  if (!context) {
+    throw new Error('useMobileNav must be used within a MobileNavProvider')
+  }
+
+  return context
+}
```

Comment:
shouldn't we keep context and hooks outside of components?

Reply:
This hook and provider component are tightly connected and the context is not exported. That's why I left it like this.